DOCS: Add Nebari destruction docs (#142)
nebari-dev/nebari-docs: docs/docs/how-tos/nebari-destroy.md

│    --disable-render                Disable auto-rendering before destroy                                                  │
│    --disable-prompt                Destroy entire Nebari cluster without confirmation request. Suggested for CI use.      │
│    --help            -h            Show this message and exit.                                                            │
╰───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╯
```

:::note
The above command will not delete your `nebar-config.yaml` or the rendered files thus a re-deployment with `nebari deploy` is possible afterwards. 
:::

Once you've decided to destroy your cluster and have read through the `nebari destroy --help` command, you are you now ready.

```bash
nebari destroy -c nebari-config.yaml
```

The terminal will prompt you to press `enter` to verify that you wish to destroy your Nebari cluster.

:::info
This destruction process can take up to 30 minutes for some cloud providers.
:::

If the destruction is successful, you will see the following output:

```bash

> [terraform]: Nebari properly destroyed all resources without error

Reviewer: While running `destroy` on an instance hosted on GCP, I remember that some "stages" failed, but that was OK because we only needed some of the stages to complete successfully -- should we consider including that here?

We can also add it later. :)

Reviewer 2: I can include a callout here 👍

```

Congratulations! You have successfully destroyed your Nebari deployment and the associated resources!

## Manually destroying the resources

### Amazon Web Services (AWS)

> Unfortunately of all the cloud providers supported, AWS seems to make it the most difficult to destroy lingering resources. They often times requiring resources be destroyed in a particular order as well.

Reviewer: ```suggestion
:::note
Destroying lingering resources on AWS can be tricky because the resources often need to be destroyed in a particular order.
:::
```

The reasoning behind this nitpick: AWS is a very popular cloud provider, and I think we may scare users by saying it's difficult.

Feel free to modify this suggestion as you see fit!

Reviewer 2: Works for me :)

#### Destroy resources from the AWS console

Here, we outline how to find and destroy resources from the AWS console. 

1. Sign into your AWS account and in the search bar type `Tag Editor`

<img src="/img/how-tos/aws_tag_editor_1.png" alt="Search for Tag Editor in the AWS console" width="700"/>

1. From the `Resource Groups & Tag Editor` page, navigate to `Tag Editor` from the left most panel

> <!-- ![Navigate to the Tag Editor in the AWS console](../../static/img/how-tos/aws_tag_editor_2.png) -->

Reviewer: I think we remove this comment because the following line adds this image?

Reviewer 2: Removed :)

   - `Regions` - the region you deployed your Nebari cluster (or select `all regions`)
   - `Resource types` - select `All supported resource types`
   - `Tags`
     - select `Environment` for key (on the left)
     - enter the namespace you chose for your Nebari cluster (on the right)

<img src="/img/how-tos/aws_tag_editor_3.png" alt="Use the Tag Editor to filter for lingering Nebari resources" width="700"/>

4. From here you can filter further if needed by entering the name of your Nebari cluster.

5. At this point, select the resource you wish to destroy. This will open another tab for you to destroy that resource.

:::info
The details on how to destroy each specific resource is beyond the scope of this guide however here is a helpful tip if you get stuck in the messy web of AWS dependencies. 

Try destroying resources like `EFS`, the `EC2 Load Balancer` or resources that might be connected to the `VPC` before deleting the `VPC` or any other network related resources.
:::

#### Destroy AWS resources using a dev tool script

This method of destroying lingering AWS resources requires cloning down the Nebari repo.

```bash
git clone https://github.com/Quansight/qhub.git
```

> Once cloned, you can a script which will attempt to destroy all lingering AWS resources.

Reviewer: ```suggestion
Once cloned, you can run a script which will attempt to destroy all lingering AWS resources.
```

```bash
doctl kubernetes cluster delete <project-name>-<namespace> -f
```

Or, if you wish to destroy your Digital Ocean resources from the Digital Ocean control panel and follow these instructions to destroy your [Kubernetes cluster](https://docs.digitalocean.com/products/kubernetes/how-to/destroy-clusters/) and these instructions to destroy the associated [space (i.e. S3)](https://docs.digitalocean.com/products/spaces/how-to/destroy/). 

### Google Cloud Platform (GCP)

If you deployed your Nebari cluster on GCP and the `nebari destroy` command failed, you will need to destroy two resources, the Kubernetes cluster (GKE) and cloud storage bucket (GCS).

You will need Google's [`gcloud`](https://cloud.google.com/sdk/gcloud) and [`gsutil`](https://cloud.google.com/storage/docs/gsutil).

To destroy the cloud storage bucket, run the following.

```bash
gsutil -m rm -r gs://<project-name>-<namespace>-terraform-state
```

And to destroy the Kubernetes cluster, run the following.

```bash
gcloud container clusters delete <project-name>-<namespace> --region <region>
```

> If you wish to destroy the cloud storage bucket from the GCP console, [follow these instructions](https://cloud.google.com/storage/docs/deleting-buckets). And to destroy the Kubernetes cluster from the GCP console, [follow these instructions](https://cloud.google.com/kubernetes-engine/docs/how-to/deleting-a-cluster#console).

Reviewer: Nitpick for accessibility, people using screen readers can benefit from explicit link texts :) 

```suggestion
If you wish to destroy the cloud storage bucket from the GCP console, [follow these instructions on Deleting buckets](https://cloud.google.com/storage/docs/deleting-buckets). And to destroy the Kubernetes cluster from the GCP console, [follow these instructions on Deleting a cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/deleting-a-cluster#console).
```

---
id: nebari-destroy
title: Destroy your Nebari deployment
description: A basic overview to destroying your Nebari deployment and associated resources
---

## Introduction

> This guide is to help first-time users destroy Nebari resources that have been previous deployed while setting up Nebari at the production scale. With your `qhub-config.yaml` configuration file, you can use a handy `destroy` command to automatically destroy resources with Nebari. We will also take a look at how you can manually destroy resources on individual cloud providers in case the automation fails.

Reviewer 3: ```suggestion
This guide is aimed to guide you through how to tear down Nebari resources previously deployed.

With your `nebari-config.yaml` configuration file, you can use a handy `destroy` command to automatically destroy resources with Nebari. We will also examine how you can manually destroy resources on individual cloud providers in case the automation fails.
```

Reviewer 3: The second sentence is a bit confusing - should be with the yaml file + the nebari CLI no?

## Destroying Nebari

> You can specify the `qhub-config.yaml` configuration file created while deploying Nebari to destroy the resources as well. Type the following command on your command line:

Reviewer 3: You can specify where?

```bash
qhub destroy -c qhub-config.yaml
```

:::note

> The above command will not delete your `qhub-config.yaml` and related rendered files thus a re-deployment via deploy is possible afterwards.

Reviewer 3: ```suggestion
The above command will not delete your `nebari-config.yaml` and related rendered files, only the cloud resources created at deploy time. Therefore, you can redeploy Nebari at a later time with the same `nebari-config.yaml` file.
```

:::

> The terminal will prompt you to press `enter` to verify that you wish to delete your Nebari deployment by rendering the resources. To disable the rendering, you can pass the flag `--disable-render` to the above command.

Reviewer 3: enter - should be using the <kbd> directive

If the deletion is successful, you will see the following output:

```bash
<!-- TODO -->

> ```

Reviewer 3: This needs to be added
